server/testing/src/main/java/org/apache/james/util/docker/DockerContainer.java

```diff
@@ -122,15 +130,16 @@ public void stop() {
     }
 
     public void pause() {
-        DockerClientFactory.instance().client().pauseContainerCmd(container.getContainerInfo().getId()).exec();
+        client().pauseContainerCmd(container.getContainerInfo().getId()).exec();
     }
 
     public boolean isRunning() {
         return container.isRunning();
     }
 
     public void unpause() {
-        DockerClientFactory.instance().client().unpauseContainerCmd(container.getContainerInfo().getId()).exec();
+        client().unpauseContainerCmd(container.getContainerInfo().getId()).exec();
+        waitStrategy.ifPresent(strategy -> strategy.waitUntilReady(container));
```

Comment:
That looks OK for a port/http strategy, however with a log waiting strategy waiting for a startup log it will not work. At all.

At the very least we should allow to bypass it if we know it won't work with the wait strategy we have. No?